acme/client.go

```diff
@@ -104,14 +104,17 @@ func (c *Client) SetChallengeProvider(challenge Challenge, p ChallengeProvider)
 		c.solvers[challenge] = &httpChallenge{jws: c.jws, validate: validate, provider: p}
 	case TLSSNI01:
 		c.solvers[challenge] = &tlsSNIChallenge{jws: c.jws, validate: validate, provider: p}
-	case DNS01:
-		c.solvers[challenge] = &dnsChallenge{jws: c.jws, validate: validate, provider: p}
 	default:
 		return fmt.Errorf("Unknown challenge %v", challenge)
 	}
 	return nil
 }
 
+func (c *Client) SetChallengeProviderDNS(challenge Challenge, p []ChallengeProvider) error {
```

Comment:
Please add a comment to exported functions. Also use plural as we're possibly setting multiple providers?